### PR TITLE
[FEATURE] 지원자 리포트 api 개발

### LINKED ISSUE #56: [FEATURE] 지원자 리포트 API 개발
### 요청 유형

> 해당되는 항목을 선택해주세요.
- [x] 새로운 기능 추가
- [ ] 기존 기능 개선
- [ ] 성능 개선

### 어떤 기능인가요?

Image

필요한 API 개발

### 작업 상세 내용

- [x] 오른쪽 카드 조회 API { 스킬핏 점수, 컬처핏 스타일, 기술스택 , 지원자 정보 (이름, 직급, 번호, 메일),메모 }
- [x] 요약 탭 조회 API
Image

- [x] 스킬핏 탭 조회 API

Image

- [x] 컬처핏 탭 조회 API

Image

- [x] 제출 서류 탭 조회 API

Image

### 유의사항

### PR DESCRIPTION
**Pull request by LeeHyunWoo02**
## 📝 PR 설명

기업 대시보드의 핵심 기능인 '지원자 상세 리포트' 화면을 구성하는 전체 API 5종을 개발했습니다.

프론트엔드에서 데이터를 별도로 가공할 필요 없이, UI 컴포넌트(방사형 차트, 카드 리스트 등)에 즉시 바인딩할 수 있도록 BFF 관점에서 DTO를 계층화하여 설계했습니다.

#### 3. [스킬핏 탭 API] - 공고의 우선순위 역량(competency_priorities)과 지원자 점수를 비교하여 맞춤형 AI 역량 요약 멘트 동적 생성

방사형(레이더) 차트 렌더링을 위한 6개 역량 퍼센트 데이터 추출

#### 4. [컬처핏 탭 API] - 슈워츠 가치관 이론에 기반한 4개 상위 축과 10개 하위 가치를 그룹핑하여 AxesDetailDTO 조립

점수가 가장 높은 상위 2개 축 자동 추출 로직 구현

#### 5. [제출서류 및 AI 면접 스크립트 탭 API]

S3 Presigned URL 기반 포트폴리오 다운로드 링크 및 외부 링크 제공

기술/인성 면접 스크립트 제공 (JSONB 배열 데이터를 InterviewScriptDTO Record 리스트로 자동 매핑)

#### 6. [테스트 코드 작성]

MatchResultReportServiceTest 작성 완료 (정상 응답 및 Null/Exception 엣지 케이스 커버리지 확보)
MatchResultReportControllerTest 작성 완료

#### PostgreSQL JSONB 데이터 매핑 처리:  면접 스크립트의 경우 List<InterviewScriptDTO> 형태의 자바 Record로 직렬화/역직렬화되도록 처리하여 서비스 로직을 극적으로 간소화했습니다.
-> jsonb 저장되는 구조가 바뀌게 된다면 추후 이슈 생성해서 jsonb 저장 구조에 맞춰서 로직 수정하겠습니다.

## 📣 Related Issue
close #56 

## 📑 P-n룰 설명
comment남기실 때 P-n규칙 꼭 남겨주세요!
P1: 꼭 반영해주세요 (Request changes)

P2: 적극적으로 고려해주세요 (Request changes)

P3: 웬만하면 반영해 주세요 (Comment)

P4: 반영해도 좋고 넘어가도 좋습니다 (Approve)

P5: 그냥 사소한 의견입니다 (Approve)

## Summary by CodeRabbit

* **New Features**
  * New report endpoints for card, AI summary, skill-fit, culture-fit and documents; interview scripts and portfolio included.
  * Applicant profile now includes position and career summary.
  * New DTOs for skill/culture details, interview scripts, portfolio and profile responses.

* **Refactor**
  * Analysis fields switched to richer map-based JSON structures; matching rate added to results.

* **Tests**
  * Added controller and service tests covering report endpoints and edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/TEAM-WEIVER/WEIVER-SERVER/pull/59)

### CODE REVIEW

File: src/main/java/com/weiver/analysis/dto/response/AnalysisReportDto.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.weiver.analysis.dto.response;
+
+import com.weiver.analysis.domain.CultureReport;
+import com.weiver.analysis.domain.TechnicalSkillReport;
+
+public record AnalysisReportDto(
+        CultureReport cultureReport,
+        TechnicalSkillReport technicalSkillReport
```
[the remark this answers is not in the document]
Comment (LeeHyunWoo02) [reply src/main/java/com/weiver/analysis/dto/response/AnalysisReportDto.java:8]: 해당 DTO는 클라이언트에게 반환할 때 사용하는 응답 DTO가 아닙니다.

File: src/main/java/com/weiver/applicant/service/WorkExperienceService.java [hunk 2]
```diff
@@ -74,6 +75,30 @@ public void updateWorkExperienceInfo(String publicId, WorkExperienceUpdateReques
...
+    /**
+     * 경력 요약 조회
+     * */
+    public List<MajorCareerDTO> getCareerSummary(String applicantPublicId) {
+        Applicant applicant = getApplicant(applicantPublicId);
+
+        List<WorkExperience> experiences = workExperienceRepository.findAllByApplicant(applicant);
+        return experiences.stream()
+                .map(MajorCareerDTO::from)
+                .toList();
```
[the remark this answers is not in the document]
Comment (LeeHyunWoo02) [reply src/main/java/com/weiver/applicant/service/WorkExperienceService.java]: 반영했습니다.

File: src/main/java/com/weiver/interview/service/InterviewSessionService.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    /**
+     * 안전한 Enum 변환 헬퍼 메서드
+     */
+    private InterviewType parseInterviewType(String typeStr) {
+        try {
+            return InterviewType.valueOf(typeStr.toUpperCase());
+        } catch (IllegalArgumentException e) {
+            throw new BusinessException(ErrorCode.INVALID_INTERVIEW_TYPE);
```
[the remark this answers is not in the document]
Comment (LeeHyunWoo02) [reply src/main/java/com/weiver/interview/service/InterviewSessionService.java:53]: NPE 방어를 위한 Null/공백 사전 검증 로직을 추가했습니다.

**Review by DongWonLee2 — APPROVED** [review]
고생하셨습니다👍
